coordinator/core/marbleapi.go

```diff
@@ -81,12 +90,38 @@ func (c *Core) Activate(ctx context.Context, req *rpc.ActivationReq) (*rpc.Activ
 	}
 
 	// customize marble's parameters
-	pemRootCA := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: c.cert.Raw})
-	pemMarbleCert := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certRaw})
-	pemMarbleKey := pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: encodedPrivKey})
-	strSealKey := hex.EncodeToString(sealKey)
+	pemRootCAObject := Secret{
+		Name:   "rootCA",
+		Public: c.cert.Raw,
+	}
+
+	pemMarbleCertObject := Secret{
+		Name:    "marbleCert",
+		Public:  certRaw,
+		Private: encodedPrivKey,
+	}
+
+	strSealKeyObject := Secret{
+		Name:    "sealKey",
+		Public:  sealKey,
+		Private: sealKey,
+	}
+
+	marbleSecrets := marbleSecretsStruct{
+		RootCA:     pemRootCAObject,
+		MarbleCert: pemMarbleCertObject,
+		SealKey:    strSealKeyObject,
+	}
```

Comment:
```suggestion
	marbleSecrets := marbleSecretsStruct{
		RootCA:     Secret{Public: c.cert.Raw},
		MarbleCert: Secret{Public: certRaw, Private: encodedPrivKey},
		SealKey:    Secret{Public: sealKey, Private: sealKey},
	}
```
I think we don't need `Name`